examples/impute/plot_multiple_imputation.py

```diff
@@ -0,0 +1,457 @@
+Both Qbar and Ubar are corrected with a factor 1 / m to account for sampling
+variance. The between imputation variance (B) is the sum of the squared
+difference between Qbar and the m point estimates, corrected with a factor
+1 / (m – 1). Then, the total variance (T) of the MI overall point estimate is
+Ubar + B + B/m.
+
+In this Example we will show how to use the `sklearn.impute.IterativeImputer`
+to perform multiple imputation. In Example 1 we show the effect of Rubin’s
+pooling rules on the variance of regression estimates. Due to the between
+imputation variance, the standard errors of all regression coefficients are
+larger with multiple imputation than with single imputation.
+
+In Example 2 we show how to set up a prediction model using multiple
+imputation. We combine the predictions from multiple imputations and show that
+this results in better regression performance.
+
+# Authors: Rianne Schouten <r.m.schouten@uu.nl>
+           Sergey Feldman <sergeyfeldman@gmail.com>
+# License: BSD 3 clause
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import stats
+
+from sklearn.datasets import load_boston
+from sklearn.linear_model import RidgeCV
+from sklearn.preprocessing import StandardScaler
+from sklearn.model_selection import train_test_split
+from sklearn.impute import IterativeImputer
+from sklearn.metrics import mean_squared_error as mse
+
+rng = np.random.RandomState(0)
+
+N_ITER = 20  # number of iterations in IterativeImputer
+N_SIM = 3  # number of simulations in Example 2
+
+
+# Start by defining a basic amputation function
+def ampute(X, missing_rate=0.75, mech="MCAR"):
+    n_samples = X.shape[0]
+    n_features = X.shape[1]
+    X_incomplete = X.copy()
+
+    n_samples_range = np.arange(n_samples)
+    size = int(missing_rate * n_samples)
+
+    # MCAR mechanism
+    if mech == 'MCAR':
+        for i in np.arange(n_features):
+            dropped_indices = np.array(np.random.choice(n_samples_range,
+                                                        size=size,
+                                                        replace=False))
+            X_incomplete[dropped_indices[:, None], i] = None
+
+    # MNAR mechanism
+    if mech == "MNAR":
+        for i in np.arange(n_features):
+            data_values = -np.mean(X[:, i]) + X[:, i]
+            weights = 1 / (1 + np.exp(-data_values))
+            probs = np.array(weights) / np.sum(np.array(weights))
+            dropped_indices = np.array(np.random.choice(n_samples_range,
+                                                        size=size,
+                                                        p=probs,
+                                                        replace=False))
+            X_incomplete[dropped_indices[:, None], i] = None
+
+    return X_incomplete
+
+
+# Make a function that calculates the variance of the beta estimates. This is
+# necessary because the linear regression model from sklearn does not provide
```

Comment:
I think we can assume without stating it that everything in the example is not in the library.